### ISSUE TITLE
please support ZhipuAI/GLM-Image

### ISSUE THREAD

**Issue by jpg1024** [Comfy-Org/ComfyUI#11857]
### Custom Node Testing

- [ ] I have tried disabling custom nodes and the issue persists (see [how to disable custom nodes](https://docs.comfy.org/troubleshooting/custom-node-issues#step-1%3A-test-with-all-custom-nodes-disabled) if you need help)

### Your question

please support ZhipuAI/GLM-Image:

[https://huggingface.co/zai-org/GLM-Image](https://huggingface.co/zai-org/GLM-Image)

thanks.

### Logs

```powershell

```

**Comment by czcz1024** [comment]
+1

**Comment by zwukong** [comment]
From the logs , not likely be supported. Meituan longcat is smaller and better,but no support either

**Comment by jpg1024** [comment]
> From the logs , not likely be supported. Meituan longcat is smaller and better,but no support either

I know it's difficult, but I want to give it a try.

**Comment by yamatazen** [comment]
SD.Next is going to support GLM-Image
https://github.com/vladmandic/sdnext/pull/4548

**Comment by zwukong** [comment]
SDNQ int4 quality  not  that good, ltx2 is very bad, z image not bad,but slower than comfyui

**Comment by CalamitousFelicitousness** [comment]
SNDQ in 4bit should be used as UINT4 with SVD enabled for optimal quality, loss wise it is nearly identical to INT8 and is better than naive FP8.

**Comment by Myoko** [comment]
I hope ComfyUI will support the GLM-Image model! it looks great.

**Comment by zwukong** [comment]
@CalamitousFelicitousness  better than naive FP8, really？ can you tell me how to enable UINT4 with SVD  in the codes.

https://huggingface.co/Disty0/GLM-Image-SDNQ-4bit-dynamic
```
import torch
import diffusers
from sdnq import SDNQConfig # import sdnq to register it into diffusers and transformers
from sdnq.common import use_torch_compile as triton_is_available
from sdnq.loader import apply_sdnq_options_to_model

pipe = diffusers.GlmImagePipeline.from_pretrained("Disty0/GLM-Image-SDNQ-4bit-dynamic", torch_dtype=torch.bfloat16, device_map="cuda")

# Enable INT8 MatMul for AMD, Intel ARC and Nvidia GPUs:
if triton_is_available and (torch.cuda.is_available() or torch.xpu.is_available()):
    pipe.transformer = apply_sdnq_options_to_model(pipe.transformer, use_quantized_matmul=True)
    # pipe.transformer = torch.compile(pipe.transformer) # optional for faster speeds

pipe.enable_model_cpu_offload()

prompt = "A beautifully designed modern food magazine style dessert recipe illustration, themed around a raspberry mousse cake. The overall layout is clean and bright, divided into four main areas: the top left features a bold black title 'Raspberry Mousse Cake Recipe Guide', with a soft-lit close-up photo of the finished cake on the right, showcasing a light pink cake adorned with fresh raspberries and mint leaves; the bottom left contains an ingredient list section, titled 'Ingredients' in a simple font, listing 'Flour 150g', 'Eggs 3', 'Sugar 120g', 'Raspberry puree 200g', 'Gelatin sheets 10g', 'Whipping cream 300ml', and 'Fresh raspberries', each accompanied by minimalist line icons (like a flour bag, eggs, sugar jar, etc.); the bottom right displays four equally sized step boxes, each containing high-definition macro photos and corresponding instructions, arranged from top to bottom as follows: Step 1 shows a whisk whipping white foam (with the instruction 'Whip egg whites to stiff peaks'), Step 2 shows a red-and-white mixture being folded with a spatula (with the instruction 'Gently fold in the puree and batter'), Step 3 shows pink liquid being poured into a round mold (with the instruction 'Pour into mold and chill for 4 hours'), Step 4 shows the finished cake decorated with raspberries and mint leaves (with the instruction 'Decorate with raspberries and mint'); a light brown information bar runs along the bottom edge, with icons on the left representing 'Preparation time: 30 minutes', 'Cooking time: 20 minutes', and 'Servings: 8'. The overall color scheme is dominated by creamy white and light pink, with a subtle paper texture in the background, featuring compact and orderly text and image layout with clear information hierarchy."
image = pipe(
    prompt=prompt,
    height=32 * 32,
    width=36 * 32,
    num_inference_steps=50,
    guidance_scale=1.5,
    generator=torch.manual_seed(42),
).images[0]
image.save("output_t2i-sdnq.png")

```

**Comment by nitinh12** [comment]
@comfyanonymous will you support this model?

**Comment by yamatazen** [comment]
> [@comfyanonymous](https://github.com/comfyanonymous) will you support this model?

We need support for autoregressive models

**Comment by diverHansun** [comment]
+1
